src/util/ActionProgramUtils.ts

```diff
@@ -239,12 +239,17 @@ export const makeAaveCloseAction = async ({
 }): Promise<ActionOp | null> => {
   const { currencyWallet: wallet } = borrowEngine
 
+  const seqAction: SeqActionOp = {
+    type: 'seq',
+    actions: []
+  }
+
   // #region Validation
 
   // Only accept this request if the user has only singular debt/collateral assets
   const collaterals = borrowEngine.collaterals.filter(collateral => !zeroString(collateral.nativeAmount))
   const debts = borrowEngine.debts.filter(debt => !zeroString(debt.nativeAmount))
-  if (debts.length === 0) return null
+  if (debts.length === 0 && collaterals.length === 0) return null
```

Comment:
Don't early exit. Check that collaterals/debts are not >1 and just return the null case at the bottom return statement.

Reply:
Looks like you left the early exit in. I'm okay with short-circuiting as a pattern when it makes sense. I've made some fixup commits for you to review or discuss and one of them is to switch from short-circuiting to nested-code. I prefer the short-circuit, but I wanted to give the nested logic a chance.